pydantic/_internal/_generate_schema.py

```diff
@@ -355,6 +356,22 @@ def _unknown_type_schema(self, obj: Any) -> CoreSchema:
             ' `__get_pydantic_core_schema__` on `<some type>` otherwise to avoid infinite recursion.'
         )
 
+    def _apply_discriminator_to_union(self, schema: CoreSchema, discriminator: Any) -> CoreSchema:
+        try:
+            return _discriminated_union.apply_discriminator(
+                schema,
+                discriminator,
+            )
+        except _discriminated_union.MissingDefinitionForUnionRef:
+            # defer until defs are resolved
+            _discriminated_union.set_discriminator(
+                schema,
+                discriminator,
+            )
+            schema.setdefault('metadata', {})[NEEDS_APPLY_DISCRIMINATED_UNION_METADATA_KEY] = True
+            self._needs_apply_discriminated_union = True
```

Comment:
I can't find any references to this field, maybe a hangover?

Reply:
I think I pushed after you last refreshed